src/inference/dev_api/openvino/runtime/threading/thread_safe_containers.hpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include <atomic>
+#include <cstddef>
+#include <mutex>
+#include <queue>
+#include <type_traits>
+
+#include "openvino/core/parallel.hpp"
```

Comment:
Why do we need to have this header in common API if only auto plugin uses it?

Reply:
ah... previously it was put in common header, may be I should move it to auto header?

Reply:
moved to auto folder

Reply:
just found that BATCH plugin is also using this, I'll move back to common folder first

Reply:
@zhaixuejun1993 I'm putting threadsafecontainer header here

Reply:
I still see this header in common part. I propose to remove it to the auto plugin.

Reply:
but this header needs to be used for batch plugin as well(https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/auto_batch/src/auto_batch.hpp#L20), not only auto plugin.